app/controllers/results_controller.rb

```diff
@@ -179,6 +179,16 @@ def run_tests
     redirect_back(fallback_location: root_path)
   end
 
+  def stop_tests
+    test_id = params[:test_run_id].to_i
+    begin
```

Comment:
Don't bother with the `rescue` here (just call `AutotestCancelJob.perform_later`)